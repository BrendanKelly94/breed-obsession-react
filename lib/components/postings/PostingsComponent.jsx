import React from 'react';
import PostComponent from './PostComponent.jsx';
import ModalComponent from './ModalComponent.jsx';
import Modal from 'react-modal';
import { Link } from 'react-router-dom';
import Loading from '../common/Loading.jsx';



class PostingsComponent extends React.Component {

  constructor(props) {
    super(props);
    this.state = { viewportWidth: document.documentElement.clientWidth };
    this.resizeCallback = this.resizeCallback.bind(this);
  }

  componentWillMount() {
    Modal.setAppElement('body');
  }

  componentWillUnmount(){
    window.removeEventListener('resize', this.resizeCallback);
  }

  componentDidMount() {
    window.addEventListener('resize', this.resizeCallback);
    this.props.forward({
      offset: this.props.offset,
      animal: this.props.animal,
      breed: this.props.breed,
      State: this.props.State,
      city: this.props.city,
      sex: this.props.sex,
      size: this.props.size,
      age: this.props.age
    });
  }

  resizeCallback(){
    this.setState({
      viewportWidth: document.documentElement.clientWidth
    });
  }

  render() {

    const {
      back, forward, selectPost, closeModal, isModalOpen,  offset, postings, selectedPost,
      animal, breed, State, city, sex,
      size, age
    } = this.props;

    const { viewportWidth } = this.state;

    let modalStyles = {
      content : {
        position: 'absolute',
        top: '40px',
        left: '5%',
        right: '5%',
        bottom: '40px',
        border: '1px solid rgb(204, 204, 204)',
        background: 'rgb(255, 255, 255)',
        overflow: 'auto',
        outline: 'none',
        padding: '20px',
        height: 'fit-content',
      }
    };

    if(viewportWidth >= 1200){
      modalStyles = {
        content : {
          position: 'absolute',
          top: '40px',
          left: '12%',
          right: '12%',
          bottom: '40px',
          border: '1px solid rgb(204, 204, 204)',
          background: 'rgb(255, 255, 255)',
          overflow: 'auto',
          outline: 'none',
          padding: '20px',
        }
      };
    }

    let forwardSearchObject = {
      offset: offset,
      animal: animal,
      breed: breed,
      State: State,
      city: city,
      sex: sex,
      size: size,
      age: age
    };

    let backSearchObject = {
      ...forwardSearchObject,
      offset: offset - (25 * 2)
    };

    return (
      <div>
        {(postings.length === 0)?
          <Loading isCenter = {true}/>

          :<div className = "container" id = "post-container">

            <Link to = "/">
              <button id = "back" className = "btn"> Home </button>
            </Link>

            {postings.map((post) => {
              return <PostComponent key = {post.id} post = {post} selectPost = {selectPost}/>;
            })
            }

            <Modal
              isOpen={isModalOpen}
              onRequestClose={closeModal}
              style={modalStyles}
              contentLabel="Postings Modal"
            >
              <ModalComponent selectedPost = {selectedPost} closeModal = {closeModal}/>
            </Modal>


            <div className = "row">
              {(offset - 25 != 0)?
                <div className = "col-lg-6" id = "back-button">
                  <button className = "btn" onClick = {() => back(backSearchObject)}> Back </button>
                </div>
                :null
              }
              <div className = "col-lg-6" id = "next-button">
                <button className = "btn" onClick = {() => forward(forwardSearchObject)}> Next </button>
              </div>
            </div>

            <footer>
              <div className = "container">
                <p>Powered by Petfinder <a href = "https:www.petfinder.com">www.petfinder.com</a></p>
              </div>
            </footer>
          </div>
        }
      </div>
    );
  }
}


export default PostingsComponent;
