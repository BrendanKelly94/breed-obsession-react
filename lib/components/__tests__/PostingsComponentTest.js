import React from 'react';
import PostingsComponent from '../postings/PostingsComponent.jsx';

import { shallow } from 'enzyme';


describe('PostingsComponent', () => {
  const testPost = {
    age:'Adult',
    breed:'Anatolian Shepherd, Akita',
    city:'Los Angeles',
    description:undefined,
    email:undefined,
    id:'40712419',
    name:'SONNY',
    phone:'(213) 485-0214',
    pics:[{'@size': 'x', '$t': 'http://photos.petfinder.com/photos/pets/40712419/1/?bust=1517392734&width=500&-x.jpg', '@id': '1'}],
    sex:'M',
    state:'CA',
    status:'A',
    street:'undefined1850 W. 60th St.',
    thumbnail:'http://photos.petfinder.com/photos/pets/40712419/1/?bust=1517392734&width=500&-x.jpg',
    zip:'90047'
  };

  const defaultTestProps = {
    selectedPost:{},
    isModalOpen:false,
    offset: 25,
    postings: [testPost],
    animal: 'dog',
    breed: 'pug',
    State: 'California',
    city: '',
    sex: '',
    size: '',
    age: '' ,
    back: jest.fn(() => ({})),
    forward: jest.fn(() => ({})),
    selectPost: jest.fn(() => ({})),
    closeModal: jest.fn(() => ({}))
  };

  const withBackButtonProps = {
    selectedPost:{},
    isModalOpen:false,
    offset: 50,
    postings: [testPost],
    animal: 'dog',
    breed: 'pug',
    State: 'California',
    city: '',
    sex: '',
    size: '',
    age: '' ,
    back: jest.fn(() => ({})),
    forward: jest.fn(() => ({})),
    selectPost: jest.fn(() => ({})),
    closeModal: jest.fn(() => ({}))
  };

  it('renders correctly', () => {
    const wrapper = shallow(
      <PostingsComponent
        {...defaultTestProps}
      />
    );

    expect(wrapper).toMatchSnapshot();

  });

  it('does not render back button when on first page', () => {
    const wrapper = shallow(
      <PostingsComponent
        {...defaultTestProps}
      />
    );

    expect(wrapper.find('#back-button').exists()).toEqual(false);
  });

  it('renders back button when not first page', () => {
    const wrapper = shallow(
      <PostingsComponent
        {...withBackButtonProps}
      />
    );

    expect(wrapper.find('#back-button').exists()).toEqual(true);
  });

});
