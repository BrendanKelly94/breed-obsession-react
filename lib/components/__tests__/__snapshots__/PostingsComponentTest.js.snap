// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PostingsComponent renders correctly 1`] = `
ShallowWrapper {
  "complexSelector": ComplexSelector {
    "buildPredicate": [Function],
    "childrenOfNode": [Function],
    "findWhereUnwrapped": [Function],
  },
  "length": 1,
  "node": <div>
    <div
      className="container"
      id="post-container"
    >
      <Link
        replace={false}
        to="/"
      >
        <button
          className="btn"
          id="back"
        >
           Home 
        </button>
      </Link>
      <PostComponent
        post={
          Object {
            "age": "Adult",
            "breed": "Anatolian Shepherd, Akita",
            "city": "Los Angeles",
            "description": undefined,
            "email": undefined,
            "id": "40712419",
            "name": "SONNY",
            "phone": "(213) 485-0214",
            "pics": Array [
              Object {
                "$t": "http://photos.petfinder.com/photos/pets/40712419/1/?bust=1517392734&width=500&-x.jpg",
                "@id": "1",
                "@size": "x",
              },
            ],
            "sex": "M",
            "state": "CA",
            "status": "A",
            "street": "undefined1850 W. 60th St.",
            "thumbnail": "http://photos.petfinder.com/photos/pets/40712419/1/?bust=1517392734&width=500&-x.jpg",
            "zip": "90047",
          }
        }
        selectPost={[MockFunction]}
      />
      <Modal
        ariaHideApp={true}
        bodyOpenClassName="ReactModal__Body--open"
        closeTimeoutMS={0}
        contentLabel="Postings Modal"
        isOpen={false}
        onRequestClose={[MockFunction]}
        parentSelector={[Function]}
        portalClassName="ReactModalPortal"
        shouldCloseOnEsc={true}
        shouldCloseOnOverlayClick={true}
        shouldFocusAfterRender={true}
        shouldReturnFocusAfterClose={true}
        style={
          Object {
            "content": Object {
              "background": "rgb(255, 255, 255)",
              "border": "1px solid rgb(204, 204, 204)",
              "bottom": "40px",
              "left": "5%",
              "outline": "none",
              "overflow": "auto",
              "padding": "20px",
              "position": "absolute",
              "right": "5%",
              "top": "40px",
            },
          }
        }
      >
        <ModalComponent
          closeModal={[MockFunction]}
          selectedPost={Object {}}
        />
      </Modal>
      <div
        className="row"
      >
        <div
          className="col-lg-6"
          id="next-button"
        >
          <button
            className="btn"
            onClick={[Function]}
          >
             Next 
          </button>
        </div>
      </div>
      <footer>
        <div
          className="container"
        >
          <p>
            Powered by Petfinder 
            <a
              href="https:www.petfinder.com"
            >
              www.petfinder.com
            </a>
          </p>
        </div>
      </footer>
    </div>
  </div>,
  "nodes": Array [
    <div>
      <div
        className="container"
        id="post-container"
      >
        <Link
          replace={false}
          to="/"
        >
          <button
            className="btn"
            id="back"
          >
             Home 
          </button>
        </Link>
        <PostComponent
          post={
            Object {
              "age": "Adult",
              "breed": "Anatolian Shepherd, Akita",
              "city": "Los Angeles",
              "description": undefined,
              "email": undefined,
              "id": "40712419",
              "name": "SONNY",
              "phone": "(213) 485-0214",
              "pics": Array [
                Object {
                  "$t": "http://photos.petfinder.com/photos/pets/40712419/1/?bust=1517392734&width=500&-x.jpg",
                  "@id": "1",
                  "@size": "x",
                },
              ],
              "sex": "M",
              "state": "CA",
              "status": "A",
              "street": "undefined1850 W. 60th St.",
              "thumbnail": "http://photos.petfinder.com/photos/pets/40712419/1/?bust=1517392734&width=500&-x.jpg",
              "zip": "90047",
            }
          }
          selectPost={[MockFunction]}
        />
        <Modal
          ariaHideApp={true}
          bodyOpenClassName="ReactModal__Body--open"
          closeTimeoutMS={0}
          contentLabel="Postings Modal"
          isOpen={false}
          onRequestClose={[MockFunction]}
          parentSelector={[Function]}
          portalClassName="ReactModalPortal"
          shouldCloseOnEsc={true}
          shouldCloseOnOverlayClick={true}
          shouldFocusAfterRender={true}
          shouldReturnFocusAfterClose={true}
          style={
            Object {
              "content": Object {
                "background": "rgb(255, 255, 255)",
                "border": "1px solid rgb(204, 204, 204)",
                "bottom": "40px",
                "left": "5%",
                "outline": "none",
                "overflow": "auto",
                "padding": "20px",
                "position": "absolute",
                "right": "5%",
                "top": "40px",
              },
            }
          }
        >
          <ModalComponent
            closeModal={[MockFunction]}
            selectedPost={Object {}}
          />
        </Modal>
        <div
          className="row"
        >
          <div
            className="col-lg-6"
            id="next-button"
          >
            <button
              className="btn"
              onClick={[Function]}
            >
               Next 
            </button>
          </div>
        </div>
        <footer>
          <div
            className="container"
          >
            <p>
              Powered by Petfinder 
              <a
                href="https:www.petfinder.com"
              >
                www.petfinder.com
              </a>
            </p>
          </div>
        </footer>
      </div>
    </div>,
  ],
  "options": Object {},
  "renderer": ReactShallowRenderer {
    "_context": Object {},
    "_element": <PostingsComponent
      State="California"
      age=""
      animal="dog"
      back={[MockFunction]}
      breed="pug"
      city=""
      closeModal={[MockFunction]}
      forward={[MockFunction]}
      isModalOpen={false}
      offset={25}
      postings={
        Array [
          Object {
            "age": "Adult",
            "breed": "Anatolian Shepherd, Akita",
            "city": "Los Angeles",
            "description": undefined,
            "email": undefined,
            "id": "40712419",
            "name": "SONNY",
            "phone": "(213) 485-0214",
            "pics": Array [
              Object {
                "$t": "http://photos.petfinder.com/photos/pets/40712419/1/?bust=1517392734&width=500&-x.jpg",
                "@id": "1",
                "@size": "x",
              },
            ],
            "sex": "M",
            "state": "CA",
            "status": "A",
            "street": "undefined1850 W. 60th St.",
            "thumbnail": "http://photos.petfinder.com/photos/pets/40712419/1/?bust=1517392734&width=500&-x.jpg",
            "zip": "90047",
          },
        ]
      }
      selectPost={[MockFunction]}
      selectedPost={Object {}}
      sex=""
      size=""
    />,
    "_forcedUpdate": false,
    "_instance": PostingsComponent {
      "context": Object {},
      "props": Object {
        "State": "California",
        "age": "",
        "animal": "dog",
        "back": [MockFunction],
        "breed": "pug",
        "city": "",
        "closeModal": [MockFunction],
        "forward": [MockFunction],
        "isModalOpen": false,
        "offset": 25,
        "postings": Array [
          Object {
            "age": "Adult",
            "breed": "Anatolian Shepherd, Akita",
            "city": "Los Angeles",
            "description": undefined,
            "email": undefined,
            "id": "40712419",
            "name": "SONNY",
            "phone": "(213) 485-0214",
            "pics": Array [
              Object {
                "$t": "http://photos.petfinder.com/photos/pets/40712419/1/?bust=1517392734&width=500&-x.jpg",
                "@id": "1",
                "@size": "x",
              },
            ],
            "sex": "M",
            "state": "CA",
            "status": "A",
            "street": "undefined1850 W. 60th St.",
            "thumbnail": "http://photos.petfinder.com/photos/pets/40712419/1/?bust=1517392734&width=500&-x.jpg",
            "zip": "90047",
          },
        ],
        "selectPost": [MockFunction],
        "selectedPost": Object {},
        "sex": "",
        "size": "",
      },
      "refs": Object {},
      "resizeCallback": [Function],
      "state": Object {
        "viewportWidth": 0,
      },
      "updater": Updater {
        "_callbacks": Array [],
        "_renderer": [Circular],
      },
    },
    "_newState": null,
    "_rendered": <div>
      <div
        className="container"
        id="post-container"
      >
        <Link
          replace={false}
          to="/"
        >
          <button
            className="btn"
            id="back"
          >
             Home 
          </button>
        </Link>
        <PostComponent
          post={
            Object {
              "age": "Adult",
              "breed": "Anatolian Shepherd, Akita",
              "city": "Los Angeles",
              "description": undefined,
              "email": undefined,
              "id": "40712419",
              "name": "SONNY",
              "phone": "(213) 485-0214",
              "pics": Array [
                Object {
                  "$t": "http://photos.petfinder.com/photos/pets/40712419/1/?bust=1517392734&width=500&-x.jpg",
                  "@id": "1",
                  "@size": "x",
                },
              ],
              "sex": "M",
              "state": "CA",
              "status": "A",
              "street": "undefined1850 W. 60th St.",
              "thumbnail": "http://photos.petfinder.com/photos/pets/40712419/1/?bust=1517392734&width=500&-x.jpg",
              "zip": "90047",
            }
          }
          selectPost={[MockFunction]}
        />
        <Modal
          ariaHideApp={true}
          bodyOpenClassName="ReactModal__Body--open"
          closeTimeoutMS={0}
          contentLabel="Postings Modal"
          isOpen={false}
          onRequestClose={[MockFunction]}
          parentSelector={[Function]}
          portalClassName="ReactModalPortal"
          shouldCloseOnEsc={true}
          shouldCloseOnOverlayClick={true}
          shouldFocusAfterRender={true}
          shouldReturnFocusAfterClose={true}
          style={
            Object {
              "content": Object {
                "background": "rgb(255, 255, 255)",
                "border": "1px solid rgb(204, 204, 204)",
                "bottom": "40px",
                "left": "5%",
                "outline": "none",
                "overflow": "auto",
                "padding": "20px",
                "position": "absolute",
                "right": "5%",
                "top": "40px",
              },
            }
          }
        >
          <ModalComponent
            closeModal={[MockFunction]}
            selectedPost={Object {}}
          />
        </Modal>
        <div
          className="row"
        >
          <div
            className="col-lg-6"
            id="next-button"
          >
            <button
              className="btn"
              onClick={[Function]}
            >
               Next 
            </button>
          </div>
        </div>
        <footer>
          <div
            className="container"
          >
            <p>
              Powered by Petfinder 
              <a
                href="https:www.petfinder.com"
              >
                www.petfinder.com
              </a>
            </p>
          </div>
        </footer>
      </div>
    </div>,
    "_rendering": false,
    "_updater": Updater {
      "_callbacks": Array [],
      "_renderer": [Circular],
    },
    "getRenderOutput": [Function],
    "render": [Function],
  },
  "root": [Circular],
  "unrendered": <PostingsComponent
    State="California"
    age=""
    animal="dog"
    back={[MockFunction]}
    breed="pug"
    city=""
    closeModal={[MockFunction]}
    forward={[MockFunction]}
    isModalOpen={false}
    offset={25}
    postings={
      Array [
        Object {
          "age": "Adult",
          "breed": "Anatolian Shepherd, Akita",
          "city": "Los Angeles",
          "description": undefined,
          "email": undefined,
          "id": "40712419",
          "name": "SONNY",
          "phone": "(213) 485-0214",
          "pics": Array [
            Object {
              "$t": "http://photos.petfinder.com/photos/pets/40712419/1/?bust=1517392734&width=500&-x.jpg",
              "@id": "1",
              "@size": "x",
            },
          ],
          "sex": "M",
          "state": "CA",
          "status": "A",
          "street": "undefined1850 W. 60th St.",
          "thumbnail": "http://photos.petfinder.com/photos/pets/40712419/1/?bust=1517392734&width=500&-x.jpg",
          "zip": "90047",
        },
      ]
    }
    selectPost={[MockFunction]}
    selectedPost={Object {}}
    sex=""
    size=""
  />,
}
`;
